homeassistant/components/synology_dsm/sensor.py

```diff
@@ -69,23 +68,22 @@ class SynoNasSensor(Entity):
     def __init__(
         self,
         api: SynoApi,
-        name: str,
         sensor_type: str,
         sensor_info: Dict[str, str],
         monitored_device: str = None,
     ):
         """Initialize the sensor."""
         self._api = api
         self.sensor_type = sensor_type
-        self._name = f"{name} {sensor_info[0]}"
+        self._name = f"{DEFAULT_NAME} {self._api.information.model} {sensor_info[0]}"
         self._unit = sensor_info[1]
         self._icon = sensor_info[2]
         self.monitored_device = monitored_device
+        self._unique_id = f"{self._api.information.serial}_{sensor_info[0]}"
 
         if self.monitored_device:
-            self._name = f"{self._name} ({self.monitored_device})"
-
-        self._unique_id = f"{self._api.information.serial} {self._name}"
```

Comment:
You can keep this as it is.
@MartinHjelmare don't want a "partly configurable" `unique_id`, as the name is not configurable now, no need to do string concat.

Reply:
It seems risky that Synology could change their name formatting in the future as then we can an unstable unique id

Reply:
I don't think a product model will change after a release!

Reply:
What if they start reporting 

DS918+
as
ds918+
or
DS918_plus

Reply:
Actually the more I think about this, you had it right the first time with the name. `self._name` should actually have the name of the device.

Its the unique_id that should not have the name which seems more inline with @MartinHjelmare  comments in the previous PR

Reply:
So we can `slugify` the `unique_id` ! :wink: 

Also if your NAS report "ds918plus" and mine "DS918+", HA will not care since it relate to our local instance of HA.

We can also see what @MartinHjelmare will say :blush:

Reply:
Just realized the api doesn't report name in information. I think its there.  May need an upstream PR

Reply:
I'm not into all the details here, but it sounds good to separate the name which can be configurable from the parts in the unique_id. The unique id have to be both unique and stable, so it must not be configurable for the same device.

Reply:
Its now separate.